src/components/structures/RoomDirectory.js

```diff
@@ -114,11 +116,21 @@ module.exports = React.createClass({
             };
         }
 
-        dis.dispatch({
-            action: 'view_room',
-            room_id: roomId,
+        var payload = {
             oob_data: oob_data,
-        });
+        };
+        if (roomAlias) {
+            payload.action = 'view_room_alias';
+            payload.room_alias = roomAlias;
```

Comment:
Actually, shouldn't this logic of 'prefer the alias if we have it' be in MatrixChat?